internals/overlord/state/state.go

```diff
@@ -293,6 +295,22 @@ func (s *State) Unlock() {
 			s.modified = false
 			return
 		}
+
+		// Let the user know what's going on if the backend writes failed.
+		if _, ok := err.(*os.PathError); ok {
```

Comment:
Why only log on PathError? Shouldn't we log in all cases? Also, let's make the log message consistent with most other log messages: "Cannot" and starting with a capital letter. Maybe `Cannot write state file, retrying: %v`.